### PR TITLE
fix ext sdcard issues with SAF while copy, move and save in text reader

### PR DESCRIPTION
**Pull request by Unpublished**
TextReader:
hide save icon after successful save
set original text only after successful save

### CODE REVIEW

File: src/main/java/com/amaze/filemanager/utils/GenericCopyUtil.java [hunk 1]
```diff
@@ -72,7 +73,18 @@ private void startCopy() throws IOException {
             } else {
 
                 // source file is neither smb nor otg; getting a channel from direct file instead of stream
-                inChannel = new RandomAccessFile(new File(mSourceFile.getPath()), "r").getChannel();
+                File file = new File(mSourceFile.getPath());
```
Comment (Unpublished) [inline src/main/java/com/amaze/filemanager/utils/GenericCopyUtil.java]: like this?

**Comment by Unpublished** [comment]
It wasn't necessary for me to move a file from ext sdcard to sdcard and you're welcome :)

**Comment by vishal0071** [comment]
If file is not readable directly from path (on sd cards), open input stream to it using ContentResolver#getInputStream. Similar to opening output stream. If a generic function already doesn't exist in FileUtil (similar to #isWritable) you could create one, or use File#canRead?

**Comment by Unpublished** [comment]
Done, check it out :)

**Pull request merged**